lib/srv/app/server.go

```diff
@@ -1187,3 +1205,13 @@ func newGetConfigForClientFn(log logrus.FieldLogger, client auth.AccessCache, tl
 		return tlsCopy, nil
 	}
 }
+
+// certFromConnState returns certificate from the connection.
+func certFromConn(tlsConn *tls.Conn) *x509.Certificate {
+	state := tlsConn.ConnectionState()
+	if len(state.PeerCertificates) != 1 {
```

Comment:
```suggestion
	if len(state.PeerCertificates) == 0 {
```

There could be a cert chain here, but I don't see why that would be a problem.